mypy/semanal.py

```diff
@@ -1851,15 +1851,15 @@ def push_type_args(
 
         return tvs
 
-    def is_defined_type_param(self, name: str) -> bool:
+    def get_defined_type_param(self, name: str) -> TypeVarLikeExpr | None:
         for names in self.locals:
```

Comment:
Ough, I spelled that out but didn't realize that this goes in the wrong direction (?). We are typically interested in the **most recent** locals frame, so shouldn't this be

```suggestion
        for names in reversed(self.locals):
```

? It didn't matter when we only requested a boolean "is defined" result, but now could potentially resolve to a wrong symbol. I don't have a testcase for that, but anyway the variable will *normally* be found in the most recent local frame, so checking it first makes more sense? Upd: no, it cannot resolve to the wrong symbol, we only accept type variables, and type param redefinitions are never added to symtable, but maybe it's still better to avoid the unnecessary work? `reversed` is fast